### PR TITLE
feat: [FFM-6911]: add 'All Environments' option to Environment selector on FF page

### PR DESCRIPTION
**Pull request by Kira-Jennings**
### Summary

- add 'All Environments' option to Environment selector on FF page
- option appears when more than one environment exists
- behind internal feature flag

#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [x] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 1]
```diff
@@ -107,18 +113,27 @@ export const useEnvironmentSelectV2 = (params: UseEnvironmentSelectV2Params) =>
...
           onChange={opt => {
             if (selectedEnvironment?.value !== opt.value) {
               setSelectedEnvironment(opt)
-              onChange(
-                opt,
-                data?.data?.content?.find(env => env.identifier === opt.value) as EnvironmentResponseDTO,
-                true
-              )
-              setActiveEnvironment(opt.value as string)
+              if (opt.value !== getString('common.allEnvironments')) {
```
Comment (Kira-Jennings) [inline src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:129]: Temporary, next PR will plumb in the new endpoint and handle the 'all' selection accordingly

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 2]
```diff
@@ -21,6 +23,7 @@ export interface UseEnvironmentSelectV2Params {
   allowCreatingNewItems?: boolean
   showCreateButton?: boolean
   noDefault?: boolean
+  allowAllOption?: boolean
```
[the remark this answers is not in the document]
Comment (Kira-Jennings) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:26]: I saw it as "allow 'All' Option"

[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:26]: gotcha, fair enough

**Review by laura-mannus — APPROVED** [review]
Nice job 🌷

**Review by crystal-fong — APPROVED** [review]
neat 🦖

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (knagurski) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:26]: You could make it super clear with something like `includeAllEnvironmentsOption`

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 3]
```diff
@@ -107,18 +113,27 @@ export const useEnvironmentSelectV2 = (params: UseEnvironmentSelectV2Params) =>
           usePortal={false}
           popoverClassName={!showCreateButton ? css.hideCreateButton : ''}
           value={selectedEnvironment}
-          items={selectOptions}
+          items={
+            ALL_ENVIRONMENTS_ENABLED && allowAllOption && selectOptions.length > 1
```
Comment (knagurski) [inline src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:117]: I think the guidance now is that we allow this view even when there is only a single env (makes no sense to me personally)
```suggestion
            ALL_ENVIRONMENTS_ENABLED && allowAllOption
```

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 4]
```diff
@@ -107,18 +113,27 @@ export const useEnvironmentSelectV2 = (params: UseEnvironmentSelectV2Params) =>
           usePortal={false}
           popoverClassName={!showCreateButton ? css.hideCreateButton : ''}
           value={selectedEnvironment}
-          items={selectOptions}
+          items={
```
Comment (knagurski) [inline src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:116]: Ideally the test for this should be in the hook's tests, not one of the pages that are using the hook

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kira-Jennings) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:26]: I'll make this change in the next PR

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 3 above)
Comment (Kira-Jennings) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:117]: I'll get confirmation on that and update if that's what they want

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 4 above)
Comment (Kira-Jennings) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:116]: Those tests don't output the proper Environment drop down element, they only check if it returns an element. I'm not touching that mess right now as it's too out of scope, however if I have more time after the first drop then I will certainly see what I can improve here

Comment (knagurski) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:116]: Yeah.. that suite (and the hook/component) is a mess and needs rethought. But it should be easy enough to get the drop down element and test it in its test suite. Deffo not a blocker... we have the test, we can refactor it into the right suite later 👍🏻

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 3 above)
Comment (knagurski) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:117]: Honestly, I think showing it when there's only a single env is pointless, so I hope they say to leave as-is 👍🏻

**Pull request merged**